hs_labels/models.py

```diff
@@ -0,0 +1,876 @@
+    """
+
+    user = models.OneToOneField(User,
+                                editable=False,
+                                null=True,
+                                related_name='ulabels',  # induced field in User class.
+                                related_query_name='ulabels')
+
+    ##########################################
+    ##########################################
+    # PUBLIC FUNCTIONS: resources
+    ##########################################
+    ##########################################
+
+    ##########################################
+    # get various aspects of labeling
+    ##########################################
+
+    @property
+    def labeled_resources(self):
+        """
+        Get a list of resources labeled by a user.
+
+        This list eliminates duplicates.
+
+        :return: List of resource objects labeled by user.
+        """
+        return BaseResource.objects.filter(rlabels__rl2url__ulabels=self,
+                                           rlabels__rl2url__kind=LabelCodes.LABEL).distinct()
+
+    @property
+    def filed_resources(self):
+        """
+        Get a list of resources filed by a user.
+
+        This list eliminates duplicates.
+
+        :return: List of resource objects assigned to folders by a user.
+        """
+        return BaseResource.objects.filter(rlabels__rl2url__ulabels=self,
+                                           rlabels__rl2url__kind=LabelCodes.FOLDER).distinct()
+
+    @property
+    def favorited_resources(self):
+        """
+        Get a list of resources favorited by a user.
+
+        This list eliminates duplicates.
+
+        :return: List of resource objects marked as favorite by a user.
+        """
+        return BaseResource.objects.filter(rlabels__rl2url__ulabels=self,
+                                           rlabels__rl2url__kind=LabelCodes.FAVORITE).distinct()
+
+    @property
+    def my_resources(self):
+        """
+        Get a list of resources marked as mine (add to my resources) by a user.
+
+        This list eliminates duplicates.
+
+        :return: List of resource objects marked as mine by a user.
+        """
+        return BaseResource.objects.filter(rlabels__rl2url__ulabels=self,
+                                           rlabels__rl2url__kind=LabelCodes.MINE).distinct()
+
+    # TODO: use access control to include accessible resources in this list.
+    @property
+    def resources_of_interest(self):
+        """
+        Get a list of resources the user has marked in any way.
+
+        :return: List of resource objects that a user has marked.
+        """
+        return BaseResource.objects.filter(rlabels__rl2url__ulabels=self).distinct()
+
+    def get_resources_with_label(self, this_label):
+        """
+        Get a list of resources with a specific label
+        """
+        if not isinstance(this_label, basestring):
+            raise HSLUsageException("Label is not text")
```

Comment:
Type checking is a python anti-pattern. Its going to fail in the regex operations in `clean_label` anyway.